### PR TITLE
cleanup asserts and enable NDEBUG

### PR DESCRIPTION
**Pull request by melvinw**

### CODE REVIEW

File: core/Makefile [hunk 1]
```diff
@@ -46,7 +46,7 @@ endif
 PROTOCXXFLAGS = -std=gnu++11 -O0
 
 CXXFLAGS += -std=gnu++11 -g3 -ggdb3 -Ofast -march=native \
-	    -Werror -isystem $(DPDK_INC_DIR) -D_GNU_SOURCE \
+	    -Werror -isystem $(DPDK_INC_DIR) -D_GNU_SOURCE -DNDEBUG \
```
Comment (barath) [inline core/Makefile]: Perhaps we can do this as a separate section, such as:

ifndef DEBUG
CXXFLAGS += -DNDEBUG
endif

**Comment by melvinw** [comment]
Yup, looks like it. Just waiting for the last g++6 build

**Comment by melvinw** [comment]
had to push one last tweak. should be ready now

**Comment by barath** [comment]
Merged, thanks!

img_0932

**Pull request merged**